### PR TITLE
fix: Mares Puck 4 descriptor match + import-duplicates design spec

### LINKED ISSUE #198: Mares Puck 4 recognized but download gets stuck at 2%
Android (Google Pixel 10 Pro Fold) syncing to a Macbook Pro:

[2026-04-11T10:02:30.864] [LDC] [ERROR] download error: Download failed [2026-04-11T10:02:30.863] [LDC] [DEBUG] nativeDownloadRun returned: -7 [2026-04-11T10:02:30.861] [BLE] [DEBUG] onConnectionStateChange: disconnected status=0 [2026-04-11T10:02:27.843] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:27.841] [BLE] [DEBUG] onCharacteristicChanged(API33+): 18 bytes [2026-04-11T10:02:27.774] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:27.764] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:23.757] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:23.755] [BLE] [DEBUG] onCharacteristicChanged(API33+): 18 bytes [2026-04-11T10:02:23.692] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:23.685] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:19.679] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:19.677] [BLE] [DEBUG] onCharacteristicChanged(API33+): 18 bytes [2026-04-11T10:02:19.610] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:19.603] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:15.597] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:15.596] [BLE] [DEBUG] onCharacteristicChanged(API33+): 18 bytes [2026-04-11T10:02:15.527] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:15.522] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:14.519] [BLE] [DEBUG] onCharacteristicChanged(API33+): 203 bytes [2026-04-11T10:02:14.463] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:14.460] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:14.457] [BLE] [DEBUG] onCharacteristicChanged(API33+): 17 bytes [2026-04-11T10:02:14.429] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:14.429] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:02:14.392] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:14.387] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:02:14.378] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:10.377] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:10.375] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:02:10.313] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:10.308] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:02:10.299] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:06.294] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:06.293] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:02:06.254] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:06.242] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:02:06.240] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:06.234] [BLE] [DEBUG] onCharacteristicChanged(API33+): 17 bytes [2026-04-11T10:02:06.208] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:06.207] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:02:06.172] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:06.166] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:02:06.160] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:02.156] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:02.155] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:02:02.115] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:02.110] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:02:02.102] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:02.101] [BLE] [DEBUG] onCharacteristicChanged(API33+): 19 bytes [2026-04-11T10:02:02.043] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:02.036] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:02:02.035] [BLE] [DEBUG] onCharacteristicChanged(API33+): 17 bytes [2026-04-11T10:02:02.007] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:02.006] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:02:01.976] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:02:01.968] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:02:01.961] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:57.957] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:57.957] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:57.895] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:57.890] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:57.882] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:53.878] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:53.876] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:53.830] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:53.825] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:53.817] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:53.816] [BLE] [DEBUG] onCharacteristicChanged(API33+): 17 bytes [2026-04-11T10:01:53.791] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:53.790] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:53.755] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:53.749] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:53.742] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:49.737] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:49.735] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:49.673] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:49.667] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:49.659] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:45.654] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:45.653] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:45.610] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:45.605] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:45.600] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:45.599] [BLE] [DEBUG] onCharacteristicChanged(API33+): 142 bytes [2026-04-11T10:01:45.541] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:45.536] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:45.536] [LDC] [DEBUG] nativeDownloadRun: vendor=Mares product=Smart model=16 name=Puck4 [2026-04-11T10:01:45.535] [BLE] [DEBUG] ensureBonded: already bonded [2026-04-11T10:01:45.535] [BLE] [DEBUG] connectAndDiscover: connected=true writeChar=99a91ebd-b21f-1689-bb43-681f1f55e966 result=true [2026-04-11T10:01:45.534] [BLE] [DEBUG] onDescriptorWrite: 00002902-0000-1000-8000-00805f9b34fb status=0 [2026-04-11T10:01:45.468] [BLE] [DEBUG] writeDescriptor returned: true [2026-04-11T10:01:45.468] [BLE] [DEBUG] CCCD descriptor: 00002902-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:45.467] [BLE] [DEBUG] write=99a91ebd-b21f-1689-bb43-681f1f55e966 notify=1d1aae28-d2a8-91a1-1242-9d2973fbe571 [2026-04-11T10:01:45.467] [BLE] [DEBUG] Data service selected (score=8) [2026-04-11T10:01:45.466] [BLE] [DEBUG] Char: d8b3ab7c-4101-ec80-c441-9b0914f6ebc3 props=0x2 descriptors=0 [2026-04-11T10:01:45.466] [BLE] [DEBUG] Char: 1d1aae28-d2a8-91a1-1242-9d2973fbe571 props=0x12 descriptors=1 [2026-04-11T10:01:45.465] [BLE] [DEBUG] Char: 99a91ebd-b21f-1689-bb43-681f1f55e966 props=0x6 descriptors=0 [2026-04-11T10:01:45.464] [BLE] [DEBUG] Service: 544e326b-5b72-c6b0-1c46-41c1bc448118 [2026-04-11T10:01:45.463] [BLE] [DEBUG] Char: 984227f3-34fc-4045-a5d0-2c581f81a153 props=0xc descriptors=0 [2026-04-11T10:01:45.463] [BLE] [DEBUG] Char: f7bf3564-fb6d-4e53-88a4-5e37e0326063 props=0x8 descriptors=0 [2026-04-11T10:01:45.462] [BLE] [DEBUG] Service: 1d14d6ee-fd63-4fa1-bfa4-8f47b42119f0 [2026-04-11T10:01:45.462] [BLE] [DEBUG] Char: 00002a23-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:45.461] [BLE] [DEBUG] Char: 00002a26-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:45.460] [BLE] [DEBUG] Char: 00002a27-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:45.459] [BLE] [DEBUG] Char: 00002a24-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:45.458] [BLE] [DEBUG] Char: 00002a29-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:45.458] [BLE] [DEBUG] Service: 0000180a-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:45.458] [BLE] [DEBUG] Char: 00002a01-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:45.457] [BLE] [DEBUG] Char: 00002a00-0000-1000-8000-00805f9b34fb props=0xa descriptors=0 [2026-04-11T10:01:45.457] [BLE] [DEBUG] Service: 00001800-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:45.457] [BLE] [DEBUG] Char: 00002b29-0000-1000-8000-00805f9b34fb props=0xa descriptors=0 [2026-04-11T10:01:45.456] [BLE] [DEBUG] Char: 00002b2a-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:45.455] [BLE] [DEBUG] Char: 00002a05-0000-1000-8000-00805f9b34fb props=0x20 descriptors=1 [2026-04-11T10:01:45.455] [BLE] [DEBUG] Service: 00001801-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:45.446] [BLE] [DEBUG] onMtuChanged: mtu=247 status=0 [2026-04-11T10:01:43.094] [LDC] [ERROR] download error: Download failed [2026-04-11T10:01:43.091] [LDC] [DEBUG] nativeDownloadRun returned: -7 [2026-04-11T10:01:40.077] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:40.076] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:40.014] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:40.009] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:40.002] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:35.996] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:35.995] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:35.936] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:35.931] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:35.923] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:31.919] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:31.918] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:31.856] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:31.847] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:31.840] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:27.838] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:27.837] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:27.772] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:27.768] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:27.758] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:23.756] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:23.755] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:23.707] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:23.703] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:23.694] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:23.693] [BLE] [DEBUG] onCharacteristicChanged(API33+): 17 bytes [2026-04-11T10:01:23.649] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:23.648] [BLE] [DEBUG] onCharacteristicChanged(API33+): 1 bytes [2026-04-11T10:01:23.613] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:23.603] [BLE] [DEBUG] write: 16 bytes, timeout=3000 [2026-04-11T10:01:23.600] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:23.595] [BLE] [DEBUG] onCharacteristicChanged(API33+): 142 bytes [2026-04-11T10:01:23.526] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:23.517] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:23.517] [LDC] [DEBUG] nativeDownloadRun: vendor=Mares product=Smart model=16 name=Puck4 [2026-04-11T10:01:23.505] [BLE] [DEBUG] ensureBonded: result=true [2026-04-11T10:01:23.499] [BLE] [DEBUG] ensureBonded: bond state changed to 12 [2026-04-11T10:01:18.078] [BLE] [DEBUG] ensureBonded: bond state changed to 11 [2026-04-11T10:01:18.026] [BLE] [DEBUG] ensureBonded: initiating bonding for 14:2D:41:ED:C6:42 [2026-04-11T10:01:18.025] [BLE] [DEBUG] connectAndDiscover: connected=true writeChar=99a91ebd-b21f-1689-bb43-681f1f55e966 result=true [2026-04-11T10:01:18.024] [BLE] [DEBUG] onDescriptorWrite: 00002902-0000-1000-8000-00805f9b34fb status=0 [2026-04-11T10:01:18.000] [BLE] [DEBUG] writeDescriptor returned: true [2026-04-11T10:01:18.000] [BLE] [DEBUG] CCCD descriptor: 00002902-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:17.999] [BLE] [DEBUG] write=99a91ebd-b21f-1689-bb43-681f1f55e966 notify=1d1aae28-d2a8-91a1-1242-9d2973fbe571 [2026-04-11T10:01:17.999] [BLE] [DEBUG] Data service selected (score=8) [2026-04-11T10:01:17.999] [BLE] [DEBUG] Char: d8b3ab7c-4101-ec80-c441-9b0914f6ebc3 props=0x2 descriptors=0 [2026-04-11T10:01:17.999] [BLE] [DEBUG] Char: 1d1aae28-d2a8-91a1-1242-9d2973fbe571 props=0x12 descriptors=1 [2026-04-11T10:01:17.999] [BLE] [DEBUG] Char: 99a91ebd-b21f-1689-bb43-681f1f55e966 props=0x6 descriptors=0 [2026-04-11T10:01:17.999] [BLE] [DEBUG] Service: 544e326b-5b72-c6b0-1c46-41c1bc448118 [2026-04-11T10:01:17.999] [BLE] [DEBUG] Char: 984227f3-34fc-4045-a5d0-2c581f81a153 props=0xc descriptors=0 [2026-04-11T10:01:17.998] [BLE] [DEBUG] Char: f7bf3564-fb6d-4e53-88a4-5e37e0326063 props=0x8 descriptors=0 [2026-04-11T10:01:17.998] [BLE] [DEBUG] Service: 1d14d6ee-fd63-4fa1-bfa4-8f47b42119f0 [2026-04-11T10:01:17.998] [BLE] [DEBUG] Char: 00002a23-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:17.998] [BLE] [DEBUG] Char: 00002a26-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:17.997] [BLE] [DEBUG] Char: 00002a27-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:17.996] [BLE] [DEBUG] Char: 00002a24-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:17.995] [BLE] [DEBUG] Char: 00002a29-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:17.995] [BLE] [DEBUG] Service: 0000180a-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:17.994] [BLE] [DEBUG] Char: 00002a01-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:17.993] [BLE] [DEBUG] Char: 00002a00-0000-1000-8000-00805f9b34fb props=0xa descriptors=0 [2026-04-11T10:01:17.990] [BLE] [DEBUG] Service: 00001800-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:17.988] [BLE] [DEBUG] Char: 00002b29-0000-1000-8000-00805f9b34fb props=0xa descriptors=0 [2026-04-11T10:01:17.986] [BLE] [DEBUG] Char: 00002b2a-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:17.985] [BLE] [DEBUG] Char: 00002a05-0000-1000-8000-00805f9b34fb props=0x20 descriptors=1 [2026-04-11T10:01:17.983] [BLE] [DEBUG] Service: 00001801-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:17.971] [BLE] [DEBUG] onMtuChanged: mtu=247 status=0 [2026-04-11T10:01:17.690] [BLE] [DEBUG] removeBond: result=true [2026-04-11T10:01:17.687] [BLE] [DEBUG] removeBond: bond state changed to 10 [2026-04-11T10:01:17.453] [BLE] [DEBUG] removeBond: removing bond for 14:2D:41:ED:C6:42 [2026-04-11T10:01:17.451] [BLE] [WARN] Auth failure (GATT status 5), removing stale bond and retrying [2026-04-11T10:01:17.451] [LDC] [DEBUG] nativeDownloadRun returned: -6 [2026-04-11T10:01:17.449] [BLE] [ERROR] write: writeCharacteristic() returned false [2026-04-11T10:01:17.448] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:13.552] [BLE] [DEBUG] onConnectionStateChange: disconnected status=5 [2026-04-11T10:01:13.444] [BLE] [DEBUG] read: size=244 timeout=3000 [2026-04-11T10:01:13.439] [BLE] [DEBUG] write: 2 bytes, timeout=3000 [2026-04-11T10:01:13.438] [LDC] [DEBUG] nativeDownloadRun: vendor=Mares product=Smart model=16 name=Puck4 [2026-04-11T10:01:13.438] [BLE] [DEBUG] ensureBonded: already bonded [2026-04-11T10:01:13.436] [BLE] [DEBUG] connectAndDiscover: connected=true writeChar=99a91ebd-b21f-1689-bb43-681f1f55e966 result=true [2026-04-11T10:01:13.435] [BLE] [DEBUG] onDescriptorWrite: 00002902-0000-1000-8000-00805f9b34fb status=0 [2026-04-11T10:01:13.403] [BLE] [DEBUG] writeDescriptor returned: true [2026-04-11T10:01:13.403] [BLE] [DEBUG] CCCD descriptor: 00002902-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:13.402] [BLE] [DEBUG] write=99a91ebd-b21f-1689-bb43-681f1f55e966 notify=1d1aae28-d2a8-91a1-1242-9d2973fbe571 [2026-04-11T10:01:13.402] [BLE] [DEBUG] Data service selected (score=8) [2026-04-11T10:01:13.401] [BLE] [DEBUG] Char: d8b3ab7c-4101-ec80-c441-9b0914f6ebc3 props=0x2 descriptors=0 [2026-04-11T10:01:13.400] [BLE] [DEBUG] Char: 1d1aae28-d2a8-91a1-1242-9d2973fbe571 props=0x12 descriptors=1 [2026-04-11T10:01:13.400] [BLE] [DEBUG] Char: 99a91ebd-b21f-1689-bb43-681f1f55e966 props=0x6 descriptors=0 [2026-04-11T10:01:13.399] [BLE] [DEBUG] Service: 544e326b-5b72-c6b0-1c46-41c1bc448118 [2026-04-11T10:01:13.399] [BLE] [DEBUG] Char: 984227f3-34fc-4045-a5d0-2c581f81a153 props=0xc descriptors=0 [2026-04-11T10:01:13.398] [BLE] [DEBUG] Char: f7bf3564-fb6d-4e53-88a4-5e37e0326063 props=0x8 descriptors=0 [2026-04-11T10:01:13.397] [BLE] [DEBUG] Service: 1d14d6ee-fd63-4fa1-bfa4-8f47b42119f0 [2026-04-11T10:01:13.397] [BLE] [DEBUG] Char: 00002a23-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:13.396] [BLE] [DEBUG] Char: 00002a26-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:13.395] [BLE] [DEBUG] Char: 00002a27-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:13.394] [BLE] [DEBUG] Char: 00002a24-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:13.394] [BLE] [DEBUG] Char: 00002a29-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:13.393] [BLE] [DEBUG] Service: 0000180a-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:13.392] [BLE] [DEBUG] Char: 00002a01-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:13.391] [BLE] [DEBUG] Char: 00002a00-0000-1000-8000-00805f9b34fb props=0xa descriptors=0 [2026-04-11T10:01:13.390] [BLE] [DEBUG] Service: 00001800-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:13.389] [BLE] [DEBUG] Char: 00002b29-0000-1000-8000-00805f9b34fb props=0xa descriptors=0 [2026-04-11T10:01:13.389] [BLE] [DEBUG] Char: 00002b2a-0000-1000-8000-00805f9b34fb props=0x2 descriptors=0 [2026-04-11T10:01:13.384] [BLE] [DEBUG] Char: 00002a05-0000-1000-8000-00805f9b34fb props=0x20 descriptors=1 [2026-04-11T10:01:13.383] [BLE] [DEBUG] Service: 00001801-0000-1000-8000-00805f9b34fb [2026-04-11T10:01:13.378] [BLE] [DEBUG] onMtuChanged: mtu=247 status=0 [2026-04-11T10:01:06.858] [BLE] [DEBUG] connectAndDiscover: connected=false writeChar=null result=false [2026-04-11T10:01:06.857] [BLE] [DEBUG] onConnectionStateChange: disconnected status=5

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

This PR bundles two independent changes that happened to land on the same local branch:

1. **Fix for #198 — Mares Puck 4 download stalls at 2%.**
   Mares Puck 4 advertises as `Puck4` (no space), but the libdivecomputer descriptor product is `Puck 4` (with space). The exact-product tiebreaker in `libdc_descriptor_match` used `strcasecmp`, so it missed, and the first family-level match won: Mares Smart (`0x10`) instead of Puck 4 (`0x35`). With the wrong model, `mares_iconhd_device_open` picked FIXED packet framing instead of the Sirius/Genius-family VARIABLE object protocol — every download stalled around 2% with `DC_STATUS_IO` (-7). Fixed by replacing the tiebreaker with a case- and space-insensitive compare so `Puck4`↔`Puck 4` (and `Quad2`↔`Quad 2`, `PuckLite`↔`Puck Lite`) match correctly.

2. **Design spec for #200 — Require explicit selection on duplicate imports.**
   New file `docs/superpowers/specs/2026-04-12-import-duplicate-required-selection-design.md`. Pre-existing local commit (authored before the Puck 4 fix); preserved in this PR rather than force-pushed into a separate branch.

## Evidence from #198
```
[LDC] [DEBUG] nativeDownloadRun: vendor=Mares product=Smart model=16 name=Puck4
[LDC] [DEBUG] nativeDownloadRun returned: -7
[LDC] [ERROR] download error: Download failed
```
`model=16` is `0x10` (SMART). With `name=Puck4`, the matcher should have resolved to `model=0x35`. After this fix it does.

## Scope of the C-side change
Only one real source file is edited: `packages/libdivecomputer_plugin/macos/Classes/libdc_wrapper.c`. All other platforms share it:
- **iOS** — symlink `ios/Classes/libdc_wrapper.c -> ../../macos/Classes/libdc_wrapper.c`
- **Android / Windows / Linux / native tests** — CMake `WRAPPER_DIR` include path

Adds a focused native regression test (`test_descriptor_matcher.c`) exercising the helper against Puck4/Puck 4, Quad2/Quad 2, PuckLite/Puck Lite plus edge cases (empty/all-space strings, case-insensitivity, prefix non-match, tab/newline non-stripping).

## Test plan
- [x] New native test `test_descriptor_matcher` passes (6/6 asserts).
- [x] Pre-push hooks pass on both commits (`dart format`, `flutter analyze`, full `flutter test` — 6327 tests).
- [ ] Manual: connect Mares Puck 4 on Android, confirm native log shows `product=Puck 4 model=53` and download completes past 2%.
- [ ] Manual: connect Mares Smart / Mares Genius and confirm regression-free auto-match.

Fixes #198

### CODE REVIEW

File: packages/libdivecomputer_plugin/macos/Classes/libdc_wrapper.c [hunk 1]
```diff
@@ -87,6 +87,24 @@ void libdc_descriptor_iterator_free(libdc_descriptor_iterator_t *iter) {
...
+static int strcasecmp_nospace(const char *a, const char *b) {
+    while (*a || *b) {
+        while (*a == ' ') a++;
+        while (*b == ' ') b++;
+        if (tolower((unsigned char)*a) != tolower((unsigned char)*b)) {
+            return 1;
+        }
+        if (*a) a++;
+        if (*b) b++;
+    }
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply packages/libdivecomputer_plugin/macos/Classes/libdc_wrapper.c]: Addressed in 76b7d65097d: tightened the helper comment to "space-insensitive" (and added explicit tab/newline non-match assertions in the test) rather than broadening the strip to isspace().

Rationale: BLE advertised names and libdivecomputer product strings don't carry tabs or newlines in practice, so matching those as spaces could only produce spurious matches, not useful ones. Keeping the helper narrow matches its purpose — bridging the exact product name ↔ BLE name difference, which is only ASCII space in every observed case (Puck4/Puck 4, Quad2/Quad 2, PuckLite/Puck Lite).

File: docs/superpowers/specs/2026-04-12-import-duplicate-required-selection-design.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+## Goal
+
+Replace the auto-skip default with an explicit "needs decision" state. Every suspected duplicate must be reviewed and resolved before the Import button becomes available. Provide efficient bulk actions so users facing many duplicates can still complete an import quickly.
+
+## Scope
+
+- Applies to the universal import flow in `lib/features/universal_import/` (the active import wizard, used by both dive-computer imports and file imports).
+- Applies to every entity type with duplicate detection: dives, sites, buddies, trips, equipment, dive centers, certifications, courses, tags, dive types.
+- Does **not** apply to the legacy `lib/features/import_wizard/` flow.
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply docs/superpowers/specs/2026-04-12-import-duplicate-required-selection-design.md:17]: Good catch. The #200 spec commit (`586bd2bae92`) was already sitting on local `main` before this fix branch was cut, so it came along for the ride. Updated the PR title and description to reflect both scopes rather than force-pushing, since the spec commit is legitimate work just waiting to land.

If a cleaner split is preferred, say the word and I'll rebase this branch onto `ca11aab0802` (the current remote main) to keep only the BLE fix, and open the spec as a separate PR.
